code/API_definitions/qod-api.yaml

```diff
@@ -325,20 +331,54 @@ components:
         * `QOS_S` - Qualifier for the requested QoS profile _S_
         * `QOS_M` - Qualifier for the requested QoS profile _M_
         * `QOS_L` - Qualifier for the requested QoS profile _L_
+    
     Notification:
       type: object
       required:
         - sessionId
-        - event
+        - eventReports
       properties:
         sessionId:
           $ref: "#/components/schemas/SessionId"
-        event:
-          $ref: "#/components/schemas/SessionEvent"
-    SessionEvent:
+        eventReports:
+          type: array
+          items:
+            $ref: "#/components/schemas/EventReport"
+          minItems: 1
+          description: Contains the reported event and applicable information
+    
+    EventReport:
+      type: object
+      required:
+        - eventType
+      properties:
+        eventType:
+          $ref: "#/components/schemas/SessionEventType"
+      discriminator:
+        propertyName: eventType
+        mapping:
+          QOS_STATUS_CHANGED: "#/components/schemas/QosStatusChangedEventReport"
```

Comment:
I believe this is still incorrect. Swagger still reports QosStatusChangedEventReport is unused. Have you tried generating code for this definition? EventReport is empty.

I would simplify this definition with something like below. Please feel free update as appropriate.

```
    EventReport:
      type: object
      required:
        - eventType
      properties:
        eventType:
          $ref: "#/components/schemas/SessionEventType"
        eventValue:
          $ref: "#/components/schemas/QosStatusChangedEventReport"
      
    QosStatusChangedEventReport:
      type: object
      required:
        - qosStatus
      properties:
        qosStatus:
          $ref: "#/components/schemas/QosStatus"
        statusInfo:
          type: string
          enum:
            - DURATION_EXPIRED
            - NETWORK_TERMINATED

    SessionEventType:
      type: string
      enum:
        - QOS_STATUS_CHANGED
```

For this definition, the json schema would be something like below (e.g.):

```
{
  "sessionId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "eventReports": [
    {
      "eventType": "QOS_STATUS_CHANGED",
      "eventValue": {
        "qosStatus": "AVAILABLE",
        "statusInfo": "DURATION_EXPIRED"
      }
    }
  ]
}
```

Reply:
@SfnUser for me it works with https://editor.swagger.io/#/, which validator are you using? Some of them don't deal well with oneOf semantics.
I think your proposal wouldn't work if we have other event types where  `eventValues` have different schemas. As commented before, we are not still there, so we can decide to simplify it for now and add the extension mechanism when needed.

Reply:
@jlurien When I made the above comment on Dec 7th, the `oneOf` semantic was not added and hence the error report was valid. However, Emil seem to have fixed that on Dec 8th with this commit https://github.com/camaraproject/QualityOnDemand/pull/67/commits/6d698c02e0050a8205f9474a41350e7599333f52 by adding `oneOf`.

The weird thing is my comment appears on that commit as well (committed on Dec 8th) but my comment is made on Dec 7th. Don't know why.

I agree that with `oneOf` addition, it works OK for me too. Cheers.

Reply:
@SfnUser Thanks for checking again. Github can not automatically evaluating if a comment is addressed by a commit and therefore keeps it. So we can resolve this conversation.